src/libraries/Directory.Build.targets

```diff
@@ -232,7 +232,10 @@
   <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->
   <Choose>
     <When Condition="'$(IncludePlatformAttributes)' == 'false'" />
-    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(IsWindowsSpecific)' == 'true') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
+    <When Condition="('$(IncludePlatformAttributes)' == 'true' or '$(SupportedOSPlatforms)' != '' or '$(UnsupportedOSPlatforms)' != '') and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

Comment:
Nit: a simpler way to do this would've been setting `IncludePlatformAttributes` to true if it was empty and any of `SupportedOSPlatforms` or `UnsupportedOSPlatforms` is not empty.

Reply:
Good idea. Is this what you were thinking, @safern?

https://github.com/dotnet/runtime/compare/main...jeffhandley:jeffhandley/use-supportedosplatforms-feedback?expand=1#diff-1a2c2a3ad9fbfb269d653ae3dc750539f97d78bb88a809e9dd08fcb04c033215

Reply:
Yeah exactly... that simplifies this hard to read condition.

Reply:
Thanks. PR incoming.